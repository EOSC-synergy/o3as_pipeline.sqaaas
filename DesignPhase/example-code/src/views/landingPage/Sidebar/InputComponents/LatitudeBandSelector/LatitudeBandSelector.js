import React from "react";
import { useDispatch } from "react-redux"
import { setLocation } from "../../../../../store/plotSlice";

/**
 * Enables the user to choose minimum and maximum latitude
 * @param {Object} props
 * @param {function} props.error - error handling
 * @returns {JSX} a JSX containing a dropdown and if "individual latitude band" is selected a number input field
 */
function LatitudeBandSelector(props) {

    const min = -90;
    const max = +90;
    const predefinedOptions = [
    {
        name: "global",
        min: 90,
        max: -90
    }
    ]

    // const dispatch = useDispatch()
    const [selectedLocations, setSelectedLocations] = React.useState(props.defaultLocation);
    let i = props.error;

    // props.allowMultipleRegions;     // TODO: how to cope with multiple possible regions in return / recovery plot?

    const handleChangeLocation = (event) => {
        // setSelectedLocations(event.target.value);
        // dispatch(setSelectedLocations(event.target.value))
    };

    return (
        <>
        </>
    );
}

export default LatitudeBandSelector;