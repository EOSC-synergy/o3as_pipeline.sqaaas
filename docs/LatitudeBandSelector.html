<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Namespace: LatitudeBandSelector</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Namespace: LatitudeBandSelector</h1>

    




<section>

<header>
    
        <h2>LatitudeBandSelector</h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

        


<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="views_landingPage_Sidebar_InputComponents_LatitudeBandSelector_LatitudeBandSelector.js.html">views/landingPage/Sidebar/InputComponents/LatitudeBandSelector/LatitudeBandSelector.js</a>, <a href="views_landingPage_Sidebar_InputComponents_LatitudeBandSelector_LatitudeBandSelector.js.html#line4">line 4</a>
    </li></ul></dd>
    

    

    

    
</dl>


        
    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id=".handleChangeLocation"><span class="type-signature">(static) </span>handleChangeLocation<span class="signature">(event)</span><span class="type-signature"></span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>event</code></td>
            

            <td class="type">
            
                
<span class="param-type">*</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="views_landingPage_Sidebar_InputComponents_LatitudeBandSelector_LatitudeBandSelector.js.html">views/landingPage/Sidebar/InputComponents/LatitudeBandSelector/LatitudeBandSelector.js</a>, <a href="views_landingPage_Sidebar_InputComponents_LatitudeBandSelector_LatitudeBandSelector.js.html#line40">line 40</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Namespaces</h3><ul><li><a href="LatitudeBandSelector.html">LatitudeBandSelector</a></li></ul><h3>Global</h3><ul><li><a href="global.html#AddModelGroupModal">AddModelGroupModal</a></li><li><a href="global.html#EditModelGroupModal">EditModelGroupModal</a></li><li><a href="global.html#ErrorMessageModal">ErrorMessageModal</a></li><li><a href="global.html#Footer">Footer</a></li><li><a href="global.html#Graph">Graph</a></li><li><a href="global.html#ModelGroupCard">ModelGroupCard</a></li><li><a href="global.html#ModelGroupConfigurator">ModelGroupConfigurator</a></li><li><a href="global.html#NavBar">NavBar</a></li><li><a href="global.html#OffsetConfigurator">OffsetConfigurator</a></li><li><a href="global.html#PlotNameField">PlotNameField</a></li><li><a href="global.html#PlotTypeSelector">PlotTypeSelector</a></li><li><a href="global.html#ReferenceYearSlider">ReferenceYearSlider</a></li><li><a href="global.html#RegionSelector">RegionSelector</a></li><li><a href="global.html#SeasonCheckBoxGroup">SeasonCheckBoxGroup</a></li><li><a href="global.html#Section">Section</a></li><li><a href="global.html#Sidebar">Sidebar</a></li><li><a href="global.html#TimeCheckBoxGroup">TimeCheckBoxGroup</a></li><li><a href="global.html#XAxisSlider">XAxisField</a></li><li><a href="global.html#YAxisSlider">YAxisSlider</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Sat Dec 18 2021 21:21:12 GMT+0100 (Mitteleurop√§ische Normalzeit)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>