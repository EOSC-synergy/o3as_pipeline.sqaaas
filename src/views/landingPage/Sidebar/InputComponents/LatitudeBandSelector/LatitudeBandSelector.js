import React from "react";
import {useDispatch, useSelector} from "react-redux"
import {selectPlotLocation, setLocation} from "../../../../../store/plotSlice/plotSlice";
import {Box, Divider, MenuItem, Select} from "@mui/material";
import Typography from "@mui/material/Typography";
import {latitudeBands} from "../../../../../utils/constants";
import PropTypes from 'prop-types';
import {fetchPlotDataForCurrentModels} from "../../../../../services/API/apiSlice";
import CustomLatitudeSelector from "./CustomLatitudeSelector/CustomLatitudeSelector";


/**
 * Enables the user to choose minimum and maximum latitude
 * @param {Object} props
 * @param {function} props.reportError - error handling
 * @returns {JSX.Element} a JSX containing a dropdown and if "individual latitude band" is selected a number input field
 */
function LatitudeBandSelector(props) {

    /**
     * A dispatch function to dispatch actions to the redux store.
     */
    const dispatch = useDispatch();

    /**
     * An object containing the current minLat and maxLat Values.
     */
    const selectedLocation = useSelector(selectPlotLocation);

    /**
     * whether the user selected to enter a custom latitude band
     */
    const [isCustomizable, setIsCustomizable] = React.useState(false);

    /**
     * handles the change when the user clicked on a new latitude band option 
     * if the user selected custom sets isCustomizable to true
     * @param {event} event the event that triggered this function call
     */
    const handleChangeLatitudeBand = (event) => {
        if (event.target.value === 'custom') {
            setIsCustomizable(true);
        } else {
            setIsCustomizable(false);
            dispatch(setLocation({minLat: event.target.value.minLat, maxLat: event.target.value.maxLat}));
            // fetch for tco3_zm and tco3_return
            dispatch(fetchPlotDataForCurrentModels());
        }
    };

    /**
     * Finds selectedLocation in latitudeBands.
     *
     * @param {boolean} forceCustomizable if true, acts like isCustomizable is true - if false, does nothing
     * @returns {{number, number}} the location
     */
    const findLatitudeBandByLocation = (forceCustomizable) => {
        if (isCustomizable || forceCustomizable) {
            return latitudeBands[latitudeBands.length - 1].value;
        }
        for (let i = 0; i < latitudeBands.length; i++) {
            if (latitudeBands[i].value.minLat === selectedLocation.minLat && latitudeBands[i].value.maxLat === selectedLocation.maxLat) {
                return latitudeBands[i].value;
            }
        }
        setIsCustomizable(true);
        findLatitudeBandByLocation(true);
    }

    return (
        <>
            <Divider>
                <Typography>LATITUDE BAND</Typography>
            </Divider>
            <Box sx={{paddingLeft: '8%', paddingRight: '8%', paddingTop: '3%'}}>
                <Select
                    sx={{width: '100%' }}
                    id="latitudeBandSelector"
                    value={findLatitudeBandByLocation(false)}
                    onChange={handleChangeLatitudeBand}
                    defaultValue={findLatitudeBandByLocation(false)}
                >
                    {
                        // maps all latitude bands from constants.js to ´MenuItem´s
                        latitudeBands.map(
                            (s, idx) => <MenuItem key={idx} value={s.value}>{s.text.description}</MenuItem>
                        )
                    }
                </Select>
                {isCustomizable && <CustomLatitudeSelector />}
            </Box>
        </>
    );
}

LatitudeBandSelector.propTypes = {
    reportError: PropTypes.func,
}

export default LatitudeBandSelector;