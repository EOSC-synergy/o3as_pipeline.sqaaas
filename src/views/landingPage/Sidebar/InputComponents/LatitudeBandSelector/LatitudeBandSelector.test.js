import React from 'react';
import ReactDOM from 'react-dom';
import LatitudeBandSelector from './LatitudeBandSelector';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect";

it('Component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LatitudeBandSelector />, div);
});

// Snapshot test
it('Component renders correctly from config file', () => {
    const tree = renderer.create(<LatitudeBandSelector />).toJSON();
    expect(tree).toMatchSnapshot();
});
