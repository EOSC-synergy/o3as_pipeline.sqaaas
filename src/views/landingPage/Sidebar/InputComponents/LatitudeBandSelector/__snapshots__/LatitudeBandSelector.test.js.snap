// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatitudeBandSelector component should render correctly from config file 1`] = `
Array [
  <div
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1ws8119-MuiDivider-root"
    role="separator"
  >
    <span
      className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        LATITUDE BAND
      </p>
    </span>
  </div>,
  <div
    className="MuiBox-root css-1n3sbe4"
  >
    <div
      className="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="latitudeBandSelectorLabel latitudeBandSelector"
          className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="latitudeBandSelector"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="notranslate"
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          disabled={false}
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value="-90,90"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1z7n62"
          >
            <span>
              LatitudeBand
            </span>
          </legend>
        </fieldset>
      </div>
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        id="latitudeBandSelectorLabel"
      >
        Latitude Band
      </label>
    </div>
  </div>,
]
`;
